src/providers/wms/qgswmsdataitems.cpp

```diff
@@ -41,6 +41,12 @@ QgsWMSConnectionItem::~QgsWMSConnectionItem()
   delete mCapabilitiesDownload;
 }
 
+void QgsWMSConnectionItem::refresh()
+{
+  mCapabilitiesDownload->setForceRefresh( true );
+  QgsDataItem::refresh();
+}
```

Comment:
Shouldn't this flag be cleared after the refresh has occurred?

Reply:
I had felt it ll be redundant to add the code for clearing the flag since the capabilities will only be fetched through refresh action only.

Reply:
ping @nyalldawson

Reply:
ping @nyalldawson

Reply:
Ok, sounds reasonable to me!